### ISSUE TITLE
`volta install <package>` should use latest LTS allowed by engines

### ISSUE THREAD

**Issue by rwjblue** [volta-cli/volta#480]
Currently, running `volta install some-package` Volta will use `some-package`'s `engines` field to decide what version of Node to pin the tool too in future invocations. Conversely, `volta install node` will **automatically** use the most recent Node LTS (not `latest`).

I believe that we should change the logic such that we default to using the most recent LTS that is allowed within the engines configuration for the package. Specifically, it is **very** common (a best practice even) that packages do not limit the upper bounds of engines because most packages work just fine across many many Node versions. This would make `volta install some-package` have matching behavior to `volta install node`.

tldr; `engines` usage is most commonly `>= 8` or something like that, `volta install` choosing the `latest` (instead of latest LTS) means that its not uncommon (once every 6 months or so as new Node major versions are released) that folks would be installing with a Node that is fundamentally less tested/bulletproof.

**Comment by Dmitry-N-Medvedev** [comment]
I might been mistaken, but
1. LTS does not give me the latest features (e.g. class instance private methods)
2. I personally do not care if the latest node's version is not yet battle tested.
